pkg/liquidity-mining/contracts/gauges/L2GaugeCheckpointer.sol

```diff
@@ -148,9 +135,28 @@ contract L2GaugeCheckpointer is IL2GaugeCheckpointer, ReentrancyGuard {
         Address.sendValue(msg.sender, address(this).balance);
     }
 
-    /**
-     * @dev See `IL2GaugeCheckpointer#getTotalBridgeCost`.
-     */
+    /// @inheritdoc IL2GaugeCheckpointer
+    function checkpointGaugesOfTypeAboveRelativeWeight(IGaugeAdder.GaugeType gaugeType, uint256 minRelativeWeight)
+        external
+        payable
+        override
+        withSupportedGaugeType(gaugeType)
+        nonReentrant
+    {
+        // solhint-disable-next-line not-rely-on-time
+        uint256 currentPeriod = _roundDownTimestamp(block.timestamp);
+
+        _checkpointGauges(gaugeType, minRelativeWeight, currentPeriod);
```

Comment:
We can also simplify the zero length check in `_checkpointGauges` while saving gas.
 
```
EnumerableSet.AddressSet storage typeGauges = _gauges[gaugeType];

uint256 totalTypeGauges = typeGauges.length();
if (totalTypeGauges == 0) {
    // Return early if there's no work to be done.
    return;
}
```